xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererShaders.cpp

```diff
@@ -46,13 +46,33 @@ void CRendererShaders::GetWeight(std::map<RenderMethod, int>& weights, const Vid
       // double copying (GPU->CPU->GPU)
       weight += 200;
   }
-  else if (av_pixel_format == AV_PIX_FMT_YUV420P ||
-    av_pixel_format == AV_PIX_FMT_YUV420P10 ||
-    av_pixel_format == AV_PIX_FMT_YUV420P16 ||
-    av_pixel_format == AV_PIX_FMT_NV12 ||
-    av_pixel_format == AV_PIX_FMT_P010 ||
-    av_pixel_format == AV_PIX_FMT_P016)
+  else if (av_pixel_format == AV_PIX_FMT_YUV420P || av_pixel_format == AV_PIX_FMT_NV12)
+  {
+    weight += 500; // single copying
+  }
+  else if (av_pixel_format == AV_PIX_FMT_YUV420P10 || av_pixel_format == AV_PIX_FMT_YUV420P16)
+  {
+    if (!DX::Windowing()->IsFormatSupport(DXGI_FORMAT_R16_UNORM, D3D11_FORMAT_SUPPORT_TEXTURE2D))
+    {
+      CLog::LogF(LOGWARNING, "Texture format DXGI_FORMAT_R16_UNORM is not supported.");
+      return;
+    }
+    weight += 500; // single copying
+  }
+  else if (av_pixel_format == AV_PIX_FMT_P010 || av_pixel_format == AV_PIX_FMT_P016)
+  {
+    if (!DX::Windowing()->IsFormatSupport(DXGI_FORMAT_R16_UNORM, D3D11_FORMAT_SUPPORT_TEXTURE2D))
+    {
+      CLog::LogF(LOGWARNING, "Texture format DXGI_FORMAT_R16_UNORM is not supported.");
+      return;
+    }
+    if (!DX::Windowing()->IsFormatSupport(DXGI_FORMAT_R16G16_UNORM, D3D11_FORMAT_SUPPORT_TEXTURE2D))
+    {
+      CLog::LogF(LOGWARNING, "Texture format DXGI_FORMAT_R16G16_UNORM is not supported.");
+      return;
+    }
```

Comment:
switch case will be more readable. please change to switch/case.

Reply:
Done!